### PR TITLE
feat(drawer): add drawer-type prop for easy drawer type switching

### LINKED ISSUE #132: change drawer type based on screen aspect ratio
Instead of checking for a maximum of minimum number of pixels I think it makes more sense to check the aspect ratio of the window, or maybe a combination of those factors. On a 9:16 desktop screen I think it makes more sense to have a temporary drawer than a persistent one.

Alternatively, we could add a `drawer-type` prop to `mdc-drawer` that accepts a string which would simplify manually switching between drawer types with user defined logic.

### PR DESCRIPTION
**Pull request by griest024**
Its a little messy right now but refactor will clean it up

closes #132

### CODE REVIEW

File: components/drawer/mdc-drawer.vue [hunk 1]
```diff
@@ -63,13 +76,22 @@ export default {
...
-      return this.permamnent || this.type === 'mdc-permanent-drawer'
+      return this.permanent || this.type === 'mdc-permanent-drawer'
     },
     isPersistent () {
       return this.persistent || this.type === 'mdc-persistent-drawer'
     },
     isTemporary () {
       return this.temporary || this.type === 'mdc-temporary-drawer'
+    },
+    isTypePermanent () {
```
Comment (stasson) [inline components/drawer/mdc-drawer.vue]: id prefer we keep the whole logic in type()

File: components/drawer/mdc-drawer.vue [hunk 2]
```diff
@@ -63,13 +76,22 @@ export default {
...
+      return this.permanent || this.drawerType === 'permanent'
+    },
+    isTypePersistent () {
```
Comment (stasson) [inline components/drawer/mdc-drawer.vue]: id prefer we keep the whole logic in type()

File: components/drawer/mdc-drawer.vue [hunk 3]
```diff
@@ -63,13 +76,22 @@ export default {
...
+      return this.persistent || this.drawerType === 'persistent'
+    },
+    isTypeTemporary () {
+      return this.temporary || this.drawerType === 'temporary'
```
Comment (stasson) [inline components/drawer/mdc-drawer.vue]: id prefer we keep the whole logic in type()

**Comment by griest024** [comment]
@stasson It's less efficient to have more complex computed properties. Every time one of the props changes, the entire property has to be recomputed. By breaking it up into more computed properties, the values can be efficiently cached.

**Pull request closed without merging**